test/Land/Model/heat_analytic_unit_test.jl

```diff
@@ -0,0 +1,167 @@
+    state::Vars,
+    aux::Vars,
+    diffusive::Vars,
+    Δx,
+    Δt,
+    t,
+    direction,
+)
+    return Δt * m.soil.param_functions.κ_dry / (Δx * Δx)
+end
+
+@testset "Heat analytic unit test" begin
+    ClimateMachine.init()
+    FT = Float32
+
+    function init_soil!(land, state, aux, coordinates, time)
+        myFT = eltype(state)
+
+        ϑ_l, θ_i = get_water_content(land.soil.water, aux, state, time)
+        θ_l =
+            volumetric_liquid_fraction(ϑ_l, land.soil.param_functions.porosity)
+        ρc_s = volumetric_heat_capacity(
+            θ_l,
+            θ_i,
+            land.soil.param_functions.ρc_ds,
+            land.param_set,
+        )
+
+        state.soil.heat.ρe_int = myFT(volumetric_internal_energy(
+            θ_i,
+            ρc_s,
+            land.soil.heat.initialT(aux),
+            land.param_set,
+        ))
+    end
+
+    soil_param_functions = SoilParamFunctions{FT}(
+        porosity = 0.495,
+        ν_gravel = 0.1,
+        ν_om = 0.1,
+        ν_sand = 0.1,
+        ρc_ds = 1,
+        κ_dry = 1,
+        κ_sat_unfrozen = 0.57,
+        κ_sat_frozen = 2.29,
+        a = 0.24,
+        b = 18.1,
+    )
+
+    heat_surface_state = (aux, t) -> eltype(aux)(0.0)
+
+    tau = FT(1) # period (sec)
+    A = FT(5) # ampltitude (T)
+    ω = FT(2 * pi / tau)
+    heat_bottom_state = (aux, t) -> A * cos(ω * t)
+    T_init = (aux) -> eltype(aux)(0.0)
+
+    soil_water_model = PrescribedWaterModel(
+        (aux, t) -> eltype(aux)(0.0),
+        (aux, t) -> eltype(aux)(0.0),
+    )
+
+    soil_heat_model = SoilHeatModel(
+        FT;
+        initialT = T_init,
+        dirichlet_bc = Dirichlet(
+            surface_state = heat_surface_state,
+            bottom_state = heat_bottom_state,
+        ),
+        neumann_bc = Neumann(surface_flux = nothing, bottom_flux = nothing),
+    )
+
+    m_soil = SoilModel(soil_param_functions, soil_water_model, soil_heat_model)
+    sources = ()
+    m = LandModel(
+        param_set,
+        m_soil;
+        source = sources,
+        init_state_prognostic = init_soil!,
+    )
+
+    N_poly = 5
+    nelem_vert = 10
+
+    # Specify the domain boundaries
+    zmax = FT(1)
+    zmin = FT(0)
+
+    driver_config = ClimateMachine.SingleStackConfiguration(
+        "LandModel",
+        N_poly,
+        nelem_vert,
+        zmax,
+        param_set,
+        m;
+        zmin = zmin,
+        numerical_flux_first_order = CentralNumericalFluxFirstOrder(),
+    )
+
+    t0 = FT(0)
+    timeend = FT(5)
+
+    solver_config = ClimateMachine.SolverConfiguration(
+        t0,
+        timeend,
+        driver_config;
+        Courant_number = FT(0.7),
+        CFL_direction = VerticalDirection(),
+    )
+    mygrid = solver_config.dg.grid
+    aux = solver_config.dg.state_auxiliary
+
+    ClimateMachine.invoke!(solver_config)
+    t = ODESolvers.gettime(solver_config.solver)
+
+    z_ind = varsindex(vars_state(m, Auxiliary(), FT), :z)
+    z = Array(aux[:, z_ind, :][:])
```

Comment:
```suggestion
    z = get_z(mygrid)
```

Reply:
Perhaps double check that `get_z` returns `z` with the same size.

Reply:
ok - will try

Reply:
Heat analytic unit test: Error During Test at C:\Users\annag\Documents\ClimateMachine.jl\test\Land\Model\heat_analytic_unit_test.jl:50
  Got exception outside of a @test
  DimensionMismatch("arrays could not be broadcast to a common size; got a dimension with lengths 60 and 2160")

Reply:
Ah, that makes sense (the previous `z` has all nodal values inside the stack). We can leave this for now, but it would be nice to clean this up at some point.

Reply:
Sounds good, I'll make a note to fix this in the next PR